lib/src/new-select/NewSelect.jsx

```diff
@@ -0,0 +1,699 @@
+            })}
+          </>
+        );
+      } else return <Option option={option} index={index} />;
+    };
+
+    return (
+      <ThemeProvider theme={colorsTheme.newSelect}>
+        <DxcSelect margin={margin} size={size} ref={ref}>
+          <Label htmlFor={selectId} disabled={disabled} backgroundType={backgroundType}>
+            {label} {optional && <OptionalLabel>(Optional)</OptionalLabel>}
+          </Label>
+          <HelperText disabled={disabled} backgroundType={backgroundType}>
+            {helperText}
+          </HelperText>
+          <SelectContainer
+            backgroundType={backgroundType}
+            disabled={disabled}
+            error={error}
+            onBlur={handleSelectOnBlur}
+            onClick={handleSelectOnClick}
+            onFocus={handleSelectOnFocus}
+            tabIndex={tabIndex}
+            ref={selectContainerRef}
+          >
+            <SearchableValueContainer>
+              {searchable && (
+                <SearchInput
+                  id={selectId}
+                  name={name}
+                  value={searchValue}
+                  disabled={disabled}
+                  onChange={handleSearchIOnChange}
+                  ref={selectInputRef}
+                  autoComplete="off"
+                  autoCorrect="off"
+                ></SearchInput>
+              )}
+              {(!searchable || searchValue === "") && (
+                <SelectedOption disabled={disabled} placeholder={!(value ?? innerValue) || isBackgroundValue}>
+                  {getSingleSelectedOptionLabel() ?? placeholder}
+                </SelectedOption>
+              )}
+            </SearchableValueContainer>
+            {!disabled && error && <ErrorIcon backgroundType={backgroundType}>{selectIcons.error}</ErrorIcon>}
+            {searchable && searchValue.length > 0 && (
+              <ClearAction onClick={handleClearActionOnClick} backgroundType={backgroundType} tabIndex={-1}>
+                {selectIcons.clear}
+              </ClearAction>
+            )}
+            <Arrow disabled={disabled} backgroundType={backgroundType}>
+              {isOpen ? selectIcons.arrowUp : selectIcons.arrowDown}
+            </Arrow>
+            {((searchable && filteredOptions.length > 0 && groupsHaveOptions() && isOpen) ||
+              (!searchable && isOpen)) && (
+              <OptionsList
+                onMouseDown={(event) => {
+                  event.preventDefault();
+                }}
+                onClick={(event) => {
+                  event.stopPropagation();
+                }}
+                onMouseLeave={() => {
+                  changeVisualFocusedSuggIndex(-1);
+                }}
+              >
+                {searchable ? filteredOptions.map(mapOptionFunc) : options.map(mapOptionFunc)}
+              </OptionsList>
+            )}
+          </SelectContainer>
+          {!disabled && <Error backgroundType={backgroundType}>{error}</Error>}
```

Comment:
Error span should be shown only if error exists, so we should add this check `{!disabled && error && <Error.../>}`. If not, the error span is always shown.